crates/core/tedge/src/cli/connect/bridge_config_aws.rs

```diff
@@ -0,0 +1,123 @@
+use crate::cli::connect::BridgeConfig;
+use tedge_config::ConnectUrl;
+use tedge_config::FilePath;
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct BridgeConfigAwsParams {
+    pub connect_url: ConnectUrl,
+    pub mqtt_tls_port: u16,
+    pub config_file: String,
+    pub remote_clientid: String,
+    pub bridge_root_cert_path: FilePath,
+    pub bridge_certfile: FilePath,
+    pub bridge_keyfile: FilePath,
+}
+
+impl From<BridgeConfigAwsParams> for BridgeConfig {
+    fn from(params: BridgeConfigAwsParams) -> Self {
+        let BridgeConfigAwsParams {
+            connect_url,
+            mqtt_tls_port,
+            config_file,
+            bridge_root_cert_path,
+            remote_clientid,
+            bridge_certfile,
+            bridge_keyfile,
+        } = params;
+
+        let address = format!("{}:{}", connect_url.as_str(), mqtt_tls_port);
+        let user_name = remote_clientid.to_string();
+        let pub_msg_topic = format!(
+            "messages/# out 1 aws/ thinedge/devices/{}/",
+            remote_clientid
+        );
+        let sub_msg_topic = format!(
+            "messages/devicebound/# in 1 aws/ thinedge/devices/{}/",
+            remote_clientid
+        );
```

Comment:
One needs to change something here.

* It sounds unusual to have inbound messages as a subset of outbound messages.
*  There is risk of an infinite loop generated by the bridge. A message published on the device to `aws/messages/devicebound/foo` will be echoed on the device to `aws/messages/devicebound/foo`. 

Why not simply a specific suffix for each direction, say `outbound` and `inbound`:

```suggestion
        let pub_msg_topic = format!(
            "messages/outbound/# out 1 aws/ thinedge/devices/{}/",
            remote_clientid
        );
        let sub_msg_topic = format!(
            "messages/inbound/# in 1 aws/ thinedge/devices/{}/",
            remote_clientid
        );
```

Reply:
As proposed by @Bravo555, it makes sense to bridge the AWS reserved `shadow` topic

```
topic # both 1 aws/device $aws/things/{client_id}/shadow
```

About measurements, alarms, and events, I would not enforce those but let the users free to add its own suffix and use the same `td` prefix as used by AWS. For inbound messages, which are commands, I would use the `cmd` prefix, again to be consistent with the AWS examples.

```
topic td/# out 1 aws/ thinedge/{device-id}/",
topic cmd/# in 1 aws/ thinedge/{device-id}/
```